Package.swift

```diff
@@ -40,6 +43,18 @@ let package = Package(
       name: "GoogleUtilities_Logger",
       dependencies: ["GoogleUtilities_Environment"],
       path: "GoogleUtilities/Logger",
-      publicHeadersPath: "Public")
+      publicHeadersPath: "Public"),
+    .target(
+      name: "FirebaseCore",
+      dependencies: ["GoogleUtilities_Environment", "GoogleUtilities_Logger"],
+      path: "Firebase/Core",
+      publicHeadersPath: "Public",
+      cSettings: [
+        .headerSearchPath("$(SRCROOT)/GoogleUtilities/Logger/Private"), // SPM doesn't support private headers
```

Comment:
I think we should add `$(SRCROOT)` only and just use paths relative to the repo root, except public headers.

This will get rid of the `SWIFT_PACKAGE` conditional imports.

Reply:
I'll try that for the intra-pod imports. 

For the cross-pod imports, I'd agree with that if we were getting rid of separate releases and versions for the Firebase pods, but there are a lot of downsides to that.

Ideally, I'd like any cross-pod dependencies to be supported properly by the package manager.

Reply:
I don't understand this comment.

Adding a header search path that points into the other module is equivalent to importing with a qualified path that points into the other module. Both of them pull the private header directly rather than from a built framework.

Secondarily, without a bunch of extra work on our part, obtaining headers from the source framework isn't any different than obtaining it from head in source: if I depend on `Foo ~> 1.2` but version 1.3 is out, Cocoapods will pull 1.3 and you won't be able to tell you've used something that was added in 1.3 and not available in 1.2.

Reply:
Agreed that what's here currently is equivalent.  The advantage of this approach is it keeps the open questions organized in the Package.swift file. 

I think it would be preferable to use module import syntax for any imports from other modules so that we're not relying on file system organization across development and deployed installations.

I'd like to investigate adding private header support to SPM or failing that, consider abandoning the use of private headers and implement them as public headers - perhaps with documentation saying they are not part of the public API.